src/client/src/tscParseOpenTSDB.c

```diff
@@ -0,0 +1,423 @@
+#include <ctype.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "hash.h"
+#include "taos.h"
+
+#include "tscUtil.h"
+#include "tsclient.h"
+#include "tscLog.h"
+
+#include "tscParseLine.h"
+
+#define MAX_TELNET_FILEDS_NUM 2
+#define OTS_TIMESTAMP_COLUMN_NAME "ts"
+#define OTS_METRIC_VALUE_COLUMN_NAME "value"
+
+/* telnet style API parser */
+static uint64_t HandleId = 0;
+
+static uint64_t genUID() {
+  uint64_t id;
+
+  do {
+    id = atomic_add_fetch_64(&HandleId, 1);
+  } while (id == 0);
+
+  return id;
+}
+
+static int32_t parseTelnetMetric(TAOS_SML_DATA_POINT *pSml, const char **index, SSmlLinesInfo* info) {
+  const char *cur = *index;
+  uint16_t len = 0;
+
+  pSml->stableName = tcalloc(TSDB_TABLE_NAME_LEN + 1, 1);    // +1 to avoid 1772 line over write
+  if (pSml->stableName == NULL){
+      return TSDB_CODE_TSC_OUT_OF_MEMORY;
+  }
+  if (isdigit(*cur)) {
+    tscError("OTD:0x%"PRIx64" Metric cannnot start with digit", info->id);
+    tfree(pSml->stableName);
+    return TSDB_CODE_TSC_LINE_SYNTAX_ERROR;
+  }
+
+  while (*cur != '\0') {
+    if (len > TSDB_TABLE_NAME_LEN) {
+      tscError("OTD:0x%"PRIx64" Metric cannot exceeds 193 characters", info->id);
+      tfree(pSml->stableName);
+      return TSDB_CODE_TSC_INVALID_TABLE_ID_LENGTH;
+    }
+
+    if (*cur == ' ') {
+      break;
+    }
+
+    pSml->stableName[len] = *cur;
+    cur++;
+    len++;
+  }
+  if (len == 0) {
+    tfree(pSml->stableName);
+    return TSDB_CODE_TSC_LINE_SYNTAX_ERROR;
+  }
+
+  pSml->stableName[len] = '\0';
+  *index = cur + 1;
```

Comment:
need to check if *cur is '\0' now.

Reply:
Done. Added check if *cur is '\0' return error since we expect more fields in the string. Also added same check logic for metric, timestamp, metric value and tag key, as these should not be the last fields in string(except for tag value).